frontend/src/components/ui/AppButton.vue

```diff
@@ -1,31 +1,44 @@
 <template>
-  <v-btn class="px-2" :class="buttonClass" v-bind="$attrs">
-    <span class="button-text overflow-hidden"><slot /></span>
-    <v-icon class="button-icon" v-if="$attrs.icon">{{ $attrs.icon }}</v-icon>
+  <v-btn class="px-2" :class="buttonClass" :size="btnSize" v-bind="$attrs">
+    <span class="overflow-hidden"><slot /></span>
+    <v-icon v-if="$attrs.icon" class="button-icon">{{ $attrs.icon }}</v-icon>
   </v-btn>
 </template>
+
 <script>
+import { isEmpty } from 'lodash'
+
 export default {
   inheritAttrs: true,
+  props: {
+    primary: {
+      type: Boolean,
+      default: true,
+    },
+    btnSize: {
```

Comment:
I believe we can simplify this by a) just reusing the existing size attribute and b) adding an additive style class to handle the sizes (e.g. small)

Reply:
it's updated. thanks for your great suggestions!